### ISSUE TITLE
Header customization

### ISSUE THREAD

**Issue by asimsir** [KleeGroup/focus-components#909]
# TL;DR
Defining a customized header seems a bit too heavy and should be done with a single public method.

# Current behaviour
Currently, if you wish to define the header part of the layout of your application, you need to manually call `dispatcher.handleViewAction()` at the end of the initialization.
For example, in `$APP_ROOT/index.js` :
```
[...]
// Démarrage de l'application
import './application';

import React from 'react';
import dispatcher from 'focus-core/dispatcher';
dispatcher.handleViewAction({
  data: {
    barContentLeftComponent: {
        component: () => <div>TEST</div>
    }
  },
  type: 'update'
});
```
This piece of code is highly inspired from the `clearHeader()` method in the `src/application/index.js` file. It will insert a `<div>TEST</div>` element in the `header-top-row-left` sub-structure.

# Suggested behaviour
Exposing a `setHeader()` method which imports custom components as arguments would make the whole process lighter. It might clearer to name arguments as the sub-structures used in the header. Something like this :
```
data: {
    headerTopRowLeft: myHeaderTopRowLeftComponent
    headerActions: myHeaderActionsComponent
    ...
}
```
The header used in the app-demo might be a good default header too (instead of an empty header).

It is also a good opportunity to migrate the `src/application` bundle to somewhere in `components`.

# Versions 
Focus-core : 0.12.3
Focus-components : 0.8.5

**Comment by pierr** [comment]
@asimsir Feel free to do a Pull Request

**Comment by pierr** [comment]
@asimsir  I think this PR is really fast to do ...

**Comment by pierr** [comment]
I also wad thinking maybe this coud be added as layout props such as an init function. It would be easier to understand.

**Comment by asimsir** [comment]
Fixed in PR  #927. Closed.
